src/Chapter2.hs

```diff
@@ -841,7 +858,12 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate num xs
+  | num < 0 = []
+  | num == 0 = xs
+  | otherwise = take (length xs) (drop num (cycle xs))
```

Comment:
Nice catch with `num == 0` 💯 

Note that you can also check `mod num (length xs)` to operate with smaller numbers in the last clause 🙂